From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Sat, 29 Jan 2022 19:28 +0000
Subject: prevent a bootmisc trying to move a nonexistant file

During boot if the "previous_dmesg" setting is enabled in
/etc/conf.d/bootmisc then during the 1st boot of a machine the
bootmisc init.d script will attempt to move a nonexistant dmesg
file, so generating an error on the console.

Modify the script to only move an existing file.

Upstream has merged this as PR 496 so it will be in the next release.

---

diff -aur a/init.d/bootmisc.in b/init.d/bootmisc.in
--- a/init.d/bootmisc.in
+++ b/init.d/bootmisc.in
@@ -226,7 +226,7 @@
 			case "$RC_SYS" in
 				VSERVER|OPENVZ|LXC|SYSTEMD-NSPAWN) ;;
 				*)
-					if yesno ${previous_dmesg:-no}; then
+					if yesno ${previous_dmesg:-no} && [ -e /var/log/dmesg ]; then
 						mv /var/log/dmesg /var/log/dmesg.old
 					fi
 					dmesg > /var/log/dmesg
